client/src/ScanPage/AddressTokensPage.tsx

```diff
@@ -1,117 +1,138 @@
-import React from 'react';
+import React, { FC, useState, useEffect } from 'react';
+import { useToasts } from 'react-toast-notifications';
+import ReactModal from 'react-modal';
 
 // routing
-import { RouteComponentProps } from 'react-router';
 import { Link } from 'react-router-dom';
 
 // libraries
 import classNames from 'classnames';
 import delve from 'dlv';
 
 /* Helpers */
-import { TokenInfo, getTokensFor, resolveENS, getENSFromAddress } from '../api';
-import { isValidAddress } from '../lib/helpers';
+import { TokenInfo, getTokensFor, resolveENS, getENSFromAddress, redeemPoaps } from '../api';
+import { isValidAddress, isValidEmail } from '../lib/helpers';
 
 /* Assets */
 import NoEventsImg from '../images/event-2019.svg';
 
 /* Components */
 import { Loading } from '../components/Loading';
+import { SubmitButton } from '../components/SubmitButton';
 
 type AddressTokensPageState = {
   tokens: null | TokenInfo[];
   address: null | string;
   ens: null | string;
   error: boolean;
   loading: boolean;
+  isRedeemModalOpen: boolean;
+  isRedeemLoading: boolean;
 };
 
-export class AddressTokensPage extends React.Component<
-  RouteComponentProps<{
-    account: string;
-  }>,
-  AddressTokensPageState
-> {
-  state: AddressTokensPageState = {
+type Props = {
+  location: any;
+  match: any;
```

Comment:
Why did we remove the proper types and switch over any? We cannot downgrade our code quality.

Reply:
I removed it to continue coding and forgot to reimplement it. Sorry!